library.js

```diff
@@ -606,6 +616,21 @@ const fetchUserInfo = async (url, token) => {
   }
 };
 
+const fetchOrganizationInfo = async (url, token) => {
+  const organizationalInfo = await fetchUserInfo(url, token);
+  if (organizationalInfo.length === 1) {
+    return organizationalInfo[0].displayName;
+  }
+  if (organizationalInfo) {
+    const primarySchoolOrganization = organizationalInfo.find(
+      (org) => org.membership && org.membership.primarySchool === true,
+    );
+    if (primarySchoolOrganization) return primarySchoolOrganization.displayName;
+    else return organizationalInfo[0].displayName;
+  }
+  return null;
```

Comment:
Er dette litt enklere å lese?

```suggestion
  const organizationalInfo = await fetchUserInfo(url, token);
  if (!organizationalInfo) return null;
  
  const primarySchoolOrganization = organizationalInfo.find(org => org.membership?.primarySchool === true);
  const primaryOrFirst = (primarySchoolOrganization ? primarySchoolOrganization : organizationalInfo[0]);
  return primaryOrFirst.displayName;

```